src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshServer.cs

```diff
@@ -136,7 +136,7 @@ public async Task WaitForClientConnectionAsync(CancellationToken cancellationTok
             {
                 while (!progressCancellationSource.Token.IsCancellationRequested)
                 {
-                    await Task.Delay(_options.RunningAsTest ? TimeSpan.MaxValue : TimeSpan.FromSeconds(5), progressCancellationSource.Token);
+                    await Task.Delay(_options.TestFlags != TestFlags.None ? TimeSpan.MaxValue : TimeSpan.FromSeconds(5), progressCancellationSource.Token);
```

Comment:
The flags also has "BrowserRequired" as an enum entry.

Reply:
It does. Any flag being set means we are running in a test, which is what is needed here.